### PR TITLE
feat: add slack webhook

### PR DESCRIPTION
**Pull request by kzuraw**
Hello 👋🏻
This is my first PR to this repository. I wanted to add support for sending slack messages when new order is created in Saleor. This PR resolves two of TODOs from #1 (the first & the second one).

## What are the changes
- feat: send slack message on `order_created` event
- docs: update README & add Slack app tutorial
	- I've updated docs based on [saleor-app-klaviyo](https://github.com/saleor/saleor-app-klaviyo)
- chore: update fields for app
	- Updated based on [saleor-app-klaviyo](https://github.com/saleor/saleor-app-klaviyo)

## How to test this PR
1. Prepare Slack app using this [doc](https://github.com/saleor/saleor-app-slack/blob/feat/add-slack-webhook/docs/setup-slack-app.md)
1. Clone this PR on your local dev machine
2. Run `pnpm i` and `pnpm run dev` + `saleor app tunnel` afterwards. More on dev commands in [Saleor docs](https://docs.saleor.io/docs/3.x/cli#app)
3. Go to your Saleor dashboard - you should be able to see `saleor-app-slack` as third party app
4. Paste your Slack webhook URL in the configuration field and save configuration
5. Create an order e.g using Saleor dashboard
6. See your message on Slack 🎉

### CODE REVIEW

File: pages/api/webhooks/order-created.ts [hunk 1]
```diff
@@ -1,23 +1,55 @@
...
+            text: `New order created for ${userEmail} 🎉\n\n<https://${saleorDomain}/dashboard/orders/${id}|View order>`,
+          },
+        },
+      ],
+    }),
+  });
+
+  if (response.status !== 200) {
+    return Response.InternalServerError({
+      success: true,
```
Comment (krzysztofwolski) [inline pages/api/webhooks/order-created.ts]:
```suggestion
      success: false,
```

File: pages/api/webhooks/order-created.ts [hunk 2]
```diff
@@ -1,23 +1,55 @@
...
+  const context = request.params;
+  const {
+    order: { id, userEmail },
+  } = context;
+
+  if (!id) {
+    return Response.BadRequest({ success: false, message: "No order id." });
+  }
+
+  const response = await fetch(webhookUrl, {
```
Comment (krzysztofwolski) [inline pages/api/webhooks/order-created.ts]: Could be moved to `/backend/slack.ts` as for example `sendSlackMessage`.

Will be easier to reuse logic and IMO keeping business logic out of the handler improve the maintenance of the app.

File: pages/api/webhooks/order-created.ts (hunk 1 above)
Comment (krzysztofwolski) [reply pages/api/webhooks/order-created.ts]: Would also add debugging logging, eg:
```
console.error("Slack API responded with code ${response.status}: ${await response.text()}")
```

**Review by krzysztofwolski — CHANGES_REQUESTED** [review]
Small changes are required, besides that - looking good! Nice setup guide 😎

File: pages/api/webhooks/order-created.ts (hunk 2 above)
Comment (kzuraw) [reply pages/api/webhooks/order-created.ts]: Sure - I was thinking if I should move to its own package but I did not do it as for the time being I don't need to call Slack API in other places. 

Anyway I will move to to its own module - what about using existing `lib` folder? I will be a bit up to date with saleor-app-klaviyo [structure](https://github.com/saleor/saleor-app-klaviyo/tree/main/lib).

Comment (krzysztofwolski) [reply pages/api/webhooks/order-created.ts]: Lib is fine too 👍

**Review by krzysztofwolski — APPROVED** [review]

**Comment by krzysztofwolski** [comment]
Was able to follow the instructions and successfully install app 🎉 
image

**Pull request merged**